lua/lspconfig/server_configurations/omnisharp.lua

```diff
@@ -42,37 +43,44 @@ return {
       return util.root_pattern '*.sln'(fname) or util.root_pattern '*.csproj'(fname)
     end,
     on_new_config = function(new_config, new_root_dir)
+      -- store the initial value of `cmd` for later use
+      if not cmd_init then
+        cmd_init = {}
+        for _, v in pairs(new_config.cmd) do
+          table.insert(cmd_init, v)
+        end
+      end
+      -- copy the fields of `cmd_init` into `new_config.cmd`
+      new_config.cmd = {}
+      for _, v in pairs(cmd_init) do
+        table.insert(new_config.cmd, v)
+      end
+      -- add hard-coded command arguments
```

Comment:
This one-liner effectively does the same thing without having to keep "global" state:

```suggestion
      new_config.cmd = { unpack(new_config.cmd or {}) }
```

Reply:
That's a great suggestion, @williamboman. I made your suggested changes, tested them, and committed them: they work great! Thanks!

Reply:
@williamboman It looks like I may have been bit hasty in my testing. While your suggestion to use `unpack` works when `cmd` is configured as a table with only one item (e.g., `cmd = { "omnisharp" }`), it fails if `cmd` needs to be instantiated in any other way.

For example, I've seen the following code suggestion in numerous places on GitHub, and it's actually included in the `docs` section of the same `omnisharp.lua` file:

``` lua
cmd = { "dotnet", "/path/to/omnisharp/OmniSharp.dll" }
```

In this instance, the line performing the `unpack` will simply assign `{ "dotnet" }` to `new_config.cmd`. Additionally, any other arguments that the end user manually passes into the program will also get dropped.

I don't see how we can accomplish what we're trying to without keeping "global" state as I had done previously, though I'm certainly no Lua expert. I'm definitely up for suggestions! And if I need to double back and stick with my original approach, I'd love to know if there's a more idiomatic way to accomplish what I was originally trying to do.

Reply:
Huh, that sounds odd. So the only thing we're trying to do here is copy a list, so no need to change approach entirely.

`new_config.cmd = vim.list_extend({}, new_config.cmd)` is an alternative, and might even be more idiomatic.

Reply:
My apologies _again_, the issue on my end was a configuration and typographical error. What I currently have committed works properly per your previous suggestion. I appreciate your patience with me!